pkg/internal/codegen/parse/util.go

```diff
@@ -395,3 +411,96 @@ func parseScaleParams(t *types.Type) (map[string]string, error) {
 	}
 	return nil, fmt.Errorf(jsonPathError)
 }
+
+// PrintColumnKV parses key-value string formatted as "foo=bar" and returns key and value.
+func PrintColumnKV(s string) (key, value string, err error) {
+	kv := strings.SplitN(s, "=", 2)
+	if len(kv) != 2 {
+		err = fmt.Errorf("invalid key value pair")
+		return key, value, err
+	}
+	key, value = kv[0], kv[1]
+	if strings.HasPrefix(value, "\"") && strings.HasSuffix(value, "\"") {
+		value = value[1 : len(value)-1]
+	}
+	return key, value, err
+}
+
+// helperPrintColumn is a helper function for the parsePrintColumnParams to compute printer columns.
+func helperPrintColumn(parts string, comment string) (v1beta1.CustomResourceColumnDefinition, error) {
+	config := v1beta1.CustomResourceColumnDefinition{}
+	var count int
+	part := strings.Split(parts, ",")
+	if len(part) < 3 {
+		return v1beta1.CustomResourceColumnDefinition{}, fmt.Errorf(printColumnError)
+	}
+
+	for _, s := range part {
+		fmt.Printf("\n[debug] %s", s)
+	}
+	for _, elem := range strings.Split(parts, ",") {
+		key, value, err := PrintColumnKV(elem)
+		if err != nil {
+			log.Fatalf("//+kubebuilder:printcolumn: tags must be key value pairs.  Expected "+
+				"keys [name=<name>,type=<type>,description=<descr>,format=<format>] "+
+				"Got string: [%s]", parts)
+		}
+		if key == "name" || key == "type" || key == "JSONPath" {
+			count++
+		}
+		switch key {
+		case printColumnName:
+			config.Name = value
+		case printColumnType:
+			if value == "integer" || value == "number" || value == "string" || value == "boolean" {
+				config.Type = value
+			} else {
+				return v1beta1.CustomResourceColumnDefinition{}, fmt.Errorf("Please enter the right value for type")
+			}
+		case printColumnFormat:
+			if config.Type == "integer" && (value == "int32" || value == "int64") {
+				config.Format = value
+			} else if config.Type == "number" && (value == "float" || value == "double") {
+				config.Format = value
+			} else if config.Type == "string" && (value == "byte" || value == "date" || value == "date-time" || value == "password") {
+				config.Format = value
+			} else {
+				return v1beta1.CustomResourceColumnDefinition{}, fmt.Errorf("Please enter the right value for format")
+			}
+		case printColumnPath:
+			config.JSONPath = value
+		case printColumnPri:
+			i, err := strconv.Atoi(value)
+			v := int32(i)
+			if err != nil {
+				return v1beta1.CustomResourceColumnDefinition{}, err
```

Comment:
we should add some context to the error in this case, something like "invalid value for priority %s for printcolumn"